drivers/pwm/pwm_mcux_sctimer.c

```diff
@@ -60,7 +60,7 @@ static int mcux_sctimer_pwm_set_cycles(const struct device *dev,
 	}
 
 	duty_cycle = 100 * pulse_cycles / period_cycles;
-
+	data->channel[channel].dutyCyclePercent = duty_cycle;
```

Comment:
This change is not needed as `channel[channel].dutyCyclePercent` is only used by `SCTIMER_SetupPwm()` function